src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -155,26 +156,17 @@ public AgentSettings LoadSettings()
             int poolId = 0;
             string agentName = null;
             WriteSection(StringUtil.Loc("RegisterAgentSectionHeader"));
-            while (true)
-            {
-                poolName = command.GetPool();
-                try
-                {
-                    poolId = await GetPoolId(poolName);
-                }
-                catch (Exception e) when (!command.Unattended)
-                {
-                    _term.WriteError(e);
-                }
 
-                if (poolId > 0)
-                {
-                    break;
-                }
-
-                _term.WriteError(StringUtil.Loc("FailedToFindPool"));
+            Type agentType = command.DeploymentAgent ? typeof(DeploymentAgentConfiguration) : typeof(AutomationAgentConfiguration);
+            var extensionManager = HostContext.GetService<IExtensionManager>();
+            IConfigurationProvider agentProvider = (extensionManager.GetExtensions<IConfigurationProvider>()).FirstOrDefault(x => x.GetType() == agentType);
+            agentProvider.InitConnection(_agentServer);
+            if (!IsHosted(serverUrl))
+            {
+                agentProvider.InitConnectionWithCollection(command, serverUrl, creds);
```

Comment:
"GetAgentQueuesAsync" does work at account level, it works at collection level that why for on-prem scenario we need to ask additional collection information